src/shared/tpm2-util.c

```diff
@@ -5233,7 +5233,7 @@ int tpm2_seal(Tpm2Context *c,
         TPMT_PUBLIC hmac_template = {
                 .type = TPM2_ALG_KEYEDHASH,
                 .nameAlg = TPM2_ALG_SHA256,
-                .objectAttributes = TPMA_OBJECT_FIXEDTPM | TPMA_OBJECT_FIXEDPARENT,
+                .objectAttributes = TPMA_OBJECT_FIXEDTPM | TPMA_OBJECT_FIXEDPARENT | (pin ? 0 : TPMA_OBJECT_NODA),
```

Comment:
so this one makes sense and afaict will not break compat. this one should go in.

Reply:
that said, could you add a brief comment here, i.e. "if protected by (user-selected and thus possibly low entropy) PIN let's enable DA, otherwise turn it off, since we then exclusively rely on strong PCR hashes for protection, which do not benefit from DA". or something like that